packages/ember-metal/lib/streams/key-stream.js

```diff
@@ -25,6 +25,12 @@ let KeyStream = BasicStream.extend({
 
   compute() {
     var object = this.sourceDep.getValue();
+    const type = typeof object;
+
+    if (type === 'boolean' || type === 'string') {
```

Comment:
@mmun you can fallback to object[key] you don't need to go into get() but I'm also thinking get() should protect an internal get() behind guards lodash style.